src/main/java/org/cloudfoundry/community/servicebroker/s3/plan/BasicPlan.java

```diff
@@ -0,0 +1,98 @@
+package org.cloudfoundry.community.servicebroker.s3.plan;
+
+import com.amazonaws.services.identitymanagement.model.AccessKey;
+import com.amazonaws.services.identitymanagement.model.User;
+import com.amazonaws.services.s3.model.Bucket;
+import org.cloudfoundry.community.servicebroker.exception.ServiceBrokerException;
+import org.cloudfoundry.community.servicebroker.model.Plan;
+import org.cloudfoundry.community.servicebroker.model.ServiceDefinition;
+import org.cloudfoundry.community.servicebroker.model.ServiceInstance;
+import org.cloudfoundry.community.servicebroker.model.ServiceInstanceBinding;
+import org.cloudfoundry.community.servicebroker.s3.service.iam.plan.BasicPlanIam;
+import org.cloudfoundry.community.servicebroker.s3.service.s3.S3;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class BasicPlan {
```

Comment:
There should probably be a `Plan` interface as all types of plans will need to implemented these methods and the Services in the `service` package shouldn't know about the different implementations.